### PR TITLE
VOL cmd sets step velocity for MIDI instruments

### PR DESCRIPTION
**Pull request by maks**
Tested using as working with a Korg NTS-1.
Includes update to the manual with photo labelling the TRS ports. 

Fixes:  #79

### CODE REVIEW

**Comment by democloid** [comment]
Seems we are mixing two concepts here, as far as I understand, volume is meant to set the instrument volume while velocity is supposed to be the specific note velocity (or volume). The original code correctly sets midi cc 7, volume. Seems we want to add a velocity command, rather than replace the original volume, no?

Part 7 here: https://www.cs.cmu.edu/~music/cmsip/readings/MIDI%20tutorial%20for%20programmers.html

**Comment by maks** [comment]
> Seems we are mixing two concepts here, as far as I understand, volume is meant to set the instrument volume while velocity is supposed to be the specific note velocity (or volume). The original code correctly sets midi cc 7, volume. Seems we want to add a velocity command, rather than replace the original volume, no?
> 
> Part 7 here: https://www.cs.cmu.edu/~music/cmsip/readings/MIDI%20tutorial%20for%20programmers.html

I think this is one area where real world practice is different from the spec. From what I've seen people expect grooveboxes to emit velocity per step and I've not seen any that send volume midi commands for that. Admittedly it may be better to have a separate command named velocity for this but I think this actually makes most sense for midi while the pico doesnt support separate velocity per step the way for example M8 does, but then the PE Tracker doesnt do a sperate per step velocity either and requires using a FX lane for that too.
I'll ask for feedback on the Discord from others to see what the expectation is around this as my view is that MIDI volumen commands are basically unused in the hardware groovebox/sequencer space.

File: sources/Application/Instruments/MidiInstrument.h [hunk 1]
```diff
@@ -61,6 +61,7 @@ class MidiInstrument : public I_Instrument {
   bool playing_;
   bool retrig_;
   int retrigLoop_;
+  char velocity_;
```
Comment (democloid) [inline sources/Application/Instruments/MidiInstrument.h:64]: set default value!

File: sources/Application/Instruments/MidiInstrument.cpp [hunk 2]
```diff
@@ -148,6 +149,11 @@ void MidiInstrument::ProcessCommand(int channel, FourCC cc, ushort value) {
     }
   } break;
 
+  case I_CMD_VELM: {
+    // VELM cmds set velocity for MIDI steps
+    velocity_ = int((value + 0.99) / 2);
```
Comment (democloid) [inline sources/Application/Instruments/MidiInstrument.cpp:154]: nit: I feel `floor(value/2)` would show intent better than this and look cleaner

**Review by democloid — CHANGES_REQUESTED** [review]

File: sources/Application/Instruments/MidiInstrument.h (hunk 1 above)
Comment (maks) [reply sources/Application/Instruments/MidiInstrument.h:64]: It's set from the instrument default value but good point! always safer to set a default value on definition.

File: sources/Application/Instruments/MidiInstrument.cpp (hunk 2 above)
Comment (maks) [reply sources/Application/Instruments/MidiInstrument.cpp:154]: I did wonder if I should change it as that's what the original lgpt code used, I'll fix it up 👍

**Review by democloid — APPROVED** [review]

**Pull request merged**